File: Data Structures & Algorithms/two-integer-sum/submission-4.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        vector<int> res;
        for(int i = 0; i<nums.size(); i++){
            if (seen.find(target - nums[i]) != seen.end()){
                res.push_back(seen[target - nums[i]]);
                res.push_back(i);
                return res;
            }
            seen[nums[i]] = i;
        }
        return res;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        l, r = 0, n - 1
        while l < n and l < r:
            lrsum = numbers[l] + numbers[r]
            if lrsum == target:
                return [l+1, r+1]
            if lrsum > target:
                r -= 1
                continue
            if lrsum < target:
                l += 1
        return []
